src/constbuffer_array.c

```diff
@@ -108,6 +108,114 @@ CONSTBUFFER_ARRAY_HANDLE constbuffer_array_create_empty(void)
     return result;
 }
 
+CONSTBUFFER_ARRAY_HANDLE constbuffer_array_create_from_array_array(const CONSTBUFFER_ARRAY_HANDLE* buffer_arrays, uint32_t buffer_array_count)
+{
+    CONSTBUFFER_ARRAY_HANDLE result;
+
+    if (
+        /*Codes_SRS_CONSTBUFFER_ARRAY_42_009: [ If buffer_arrays is NULL and buffer_array_count is not 0 then constbuffer_array_create_from_array_array shall fail and return NULL. ]*/
+        (buffer_arrays == NULL && buffer_array_count != 0)
+        )
+    {
+        LogError("invalid arguments: const CONSTBUFFER_ARRAY_HANDLE* buffer_arrays=%p, uint32_t buffer_array_count=%" PRIu32, buffer_arrays, buffer_array_count);
+        result = NULL;
+    }
+    else
+    {
+        /*Codes_SRS_CONSTBUFFER_ARRAY_42_001: [ If buffer_arrays is NULL or buffer_array_count is 0 then constbuffer_array_create_from_array_array shall create a new, empty CONSTBUFFER_ARRAY_HANDLE. ]*/
+        if (buffer_arrays == NULL || buffer_array_count == 0)
+        {
+            result = constbuffer_array_create_empty();
```

Comment:
>constbuffer_array_create_empty [](start = 21, length = 30)

Consider LogError if this fails.

Reply:
added

---
In reply to: [245157725](https://github.com/Azure/azure-c-shared-utility/pull/248#discussion_r245157725) [](ancestors = 245157725)